internal/commands/accesslist/delete.go

```diff
@@ -0,0 +1,191 @@
+		cli.ProjectFlag(&cmd.inputs.Project),
+		cli.ProductFlag(&cmd.inputs.Products),
+		flags.StringSliceFlag{
+			Value: &cmd.inputs.Addresses,
+			Meta: flags.Meta{
+				Name: "ip",
+				Usage: flags.Usage{
+					Description: "Specify the IP address(es) or CIDR block(s) to delete",
+				},
+			},
+		},
+	}
+}
+
+// Inputs is the command inputs
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+// Handler is the command handler
+func (cmd *CommandDelete) Handler(profile *user.Profile, ui terminal.UI, clients cli.Clients) error {
+	app, err := cli.ResolveApp(ui, clients.Realm, cmd.inputs.Filter())
+	if err != nil {
+		return err
+	}
+
+	allowedIPs, err := clients.Realm.AllowedIPs(app.GroupID, app.ID)
+	if err != nil {
+		return err
+	}
+
+	selectedAllowedIPs, err := cmd.inputs.resolveAllowedIP(ui, allowedIPs)
+	if err != nil {
+		return err
+	}
+
+	if len(selectedAllowedIPs) == 0 {
+		return errors.New("no IP addresses or CIDR blocks to delete")
+	}
+
+	outputs := make([]deleteAllowedIPOutput, len(selectedAllowedIPs))
+	for i, allowedIP := range selectedAllowedIPs {
+		err := clients.Realm.AllowedIPDelete(app.GroupID, app.ID, allowedIP.ID)
+		outputs[i] = deleteAllowedIPOutput{allowedIP, err}
+	}
+
+	tableRows := make([]map[string]interface{}, 0, len(outputs))
+	for _, output := range outputs {
+		row := map[string]interface{}{
+			headerAddress: output.allowedIP.Address,
+			headerComment: output.allowedIP.Comment,
+		}
+		var deleted bool
+		var err error
+		if output.err != nil {
+			err = output.err
+		} else {
+			deleted = true
+		}
+
+		row[headerDeleted] = deleted
+		row[headerDetails] = err
+		tableRows = append(tableRows, row)
+	}
+
+	ui.Print(terminal.NewTableLog(
+		fmt.Sprintf("Deleted %d IP address(es) and/or CIDR block(s)", len(outputs)),
+		deleteTableHeaders,
+		tableRows...,
+	))
+
+	return nil
+}
+
+func (i *deleteInputs) Resolve(profile *user.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory, false); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (i *deleteInputs) resolveAllowedIP(ui terminal.UI, allowedIPs []realm.AllowedIP) ([]realm.AllowedIP, error) {
+	if len(allowedIPs) == 0 {
+		return nil, nil
+	}
+
+	if len(i.Addresses) > 0 {
+		allowedIPsByAddress := make(map[string]realm.AllowedIP, len(allowedIPs))
+		for _, allowedIP := range allowedIPs {
+			allowedIPsByAddress[allowedIP.Address] = allowedIP
+		}
+
+		allowedIPs := make([]realm.AllowedIP, 0, len(i.Addresses))
+		var notFoundAddresses []string
+		for _, address := range i.Addresses {
+			if allowedIP, ok := allowedIPsByAddress[address]; ok {
+				allowedIPs = append(allowedIPs, allowedIP)
+			} else {
+				notFoundAddresses = append(notFoundAddresses, address)
+			}
+		}
+
+		if len(notFoundAddresses) > 0 {
+			return nil, errors.New(
+				"unable to find IP address(es) and/or CIDR block(s): " +
+					strings.Join(notFoundAddresses, ", "),
+			)
+		}
+
+		return allowedIPs, nil
```

Comment:
I think this would be more clear if we also named this `selectedAllowedIPs`.  Also I just don't like the feeling of overwriting `allowedIPs` with a new data structure.